modules/security-jwt/src/main/java/com/devonfw/module/security/jwt/util/TokenExtractor.java

```diff
@@ -1,140 +1,104 @@
+    Jwt jwt = JwtHelper.decodeAndVerify(token, verifier);
 
-			if(tokenClaims.get("nbf")!=null) {
-				Instant nbf = Instant.ofEpochSecond(((Number) tokenClaims.get("nbf")).longValue());
-				if (now.isBefore(nbf)) {
-					// Throw exception
-					LOG.info("Token is not valid before " + nbf);
-				}
+    String claims = jwt.getClaims();
+    try {
+      this.tokenClaims = this.objectMapper.readValue(claims, Map.class);
 
-			}
+      LOG.info("Token Claims " + this.tokenClaims.toString());
 
-			if(tokenClaims.get("exp") !=null) {
-				Instant expiration_time = Instant.ofEpochSecond(((Number) tokenClaims.get("exp")).longValue());
+      Instant now = Instant.now();
 
-				if (now.isAfter(expiration_time)) {
-					// Throw exception
-					LOG.info("Token is expired");
-				}
-			}
+      Object nbfObj = this.tokenClaims.get("nbf");
+      Object expObj = this.tokenClaims.get("exp");
 
+      if (nbfObj != null) {
+        Instant nbf = Instant.ofEpochSecond(((Number) nbfObj).longValue());
+        if (now.isBefore(nbf)) {
+          throw new IllegalStateException("Token is not valid before " + nbf);
+        }
 
-			String user = tokenClaims.get("sub").toString();
-			// If token do not throw any exception above it is valid and we need to create
-			// authentication object from it
+      }
 
-			authentication = new UsernamePasswordAuthenticationToken(user,
-					tokenClaims.get("roles").toString());
-		} catch (IOException e) {
+      if (expObj != null) {
+        Instant expiration_time = Instant.ofEpochSecond(((Number) expObj).longValue());
 
-			LOG.error(e.getMessage());
-		}
-		return authentication;
-	}
+        if (now.isAfter(expiration_time)) {
+          throw new IllegalStateException("Token is expired" + expiration_time);
+        }
+      }
 
-	private String getAlgorithmFamilyType(String algorithms) {
-		StringBuilder familyAlgorithm = null;
-		List<SignatureAlgorithm> algos = this.keyStoreAccess.loadAllAlgorithmList();
-		for (SignatureAlgorithm algorithm : algos) {
-			if (algorithm.getValue().equalsIgnoreCase(algorithms)) {
-				LOG.info("Algorithm exists--");
-				familyAlgorithm = new StringBuilder(algorithm.getFamilyName());
-				LOG.info("Parent algo is " + familyAlgorithm.toString());
-				return familyAlgorithm.toString();
-			}
-		}
-		return null;
-	}
+      String user = this.tokenClaims.get("sub").toString();
 
-	public SignatureVerifier signatureVerifierFactory(PublicKey publicKey, String algorithm) {
-		switch (algorithm) {
-		case "Elliptic Curve":
-			return null;
+      List<String> roles = (List<String>) this.tokenClaims.get("roles");
 
-		case "HMAC":
-			return null;
+      List<GrantedAuthority> authorities = new ArrayList<>();
+      for (String role : roles) {
+        authorities.add(new SimpleGrantedAuthority(role));
+      }
 
-		case "RSA":
-			return new RsaVerifier((RSAPublicKey) publicKey);
+      authentication = new UsernamePasswordAuthenticationToken(user, null, authorities);
```

Comment:
The fabrication of the Authentication should go to a replacable component or at least a protected method that can be overridden. Projects will often require custom code here and might have their own custom implementation of Authentication that needs to be used.